dbms/src/Storages/DeltaMerge/Delta/Snapshot.cpp

```diff
@@ -39,9 +40,10 @@ DeltaSnapshotPtr DeltaValueSpace::createSnapshot(const DMContext & context, bool
     snap->delta = this->shared_from_this();
 
     auto storage_snap = std::make_shared<StorageSnapshot>(context.storage_pool, context.getReadLimiter(), context.tracing_id, /*snapshot_read*/ true);
-    snap->persisted_files_snap = persisted_file_set->createSnapshot(storage_snap);
+    auto read_from_storage_snap = ColumnFileDataProviderLocalStoragePool::create(storage_snap);
+    snap->persisted_files_snap = persisted_file_set->createSnapshot(read_from_storage_snap);
+    snap->mem_table_snap = mem_table_set->createSnapshot(read_from_storage_snap, for_update);
```

Comment:
```suggestion
    auto data_from_storage_snap = ColumnFileDataProviderLocalStoragePool::create(storage_snap);
    snap->persisted_files_snap = persisted_file_set->createSnapshot(data_from_storage_snap);
    snap->mem_table_snap = mem_table_set->createSnapshot(data_from_storage_snap, for_update);
```
`read_from_storage_snap` sounds like a function name to me, `data_from_storage_snap` is better